landlab/components/carbonate/carbonate_producer.py

```diff
@@ -0,0 +1,219 @@
+#! /usr/bin/env python
+from contextlib import contextmanager
+
+import numpy as np
+from landlab import Component
+
+
+@contextmanager
+def set_numpy_err(*args, **kwds):
+    settings = np.seterr(*args, **kwds)
+    try:
+        yield settings
+    finally:
+        np.seterr(**settings)
+
+
+def smooth_heaviside(x, width=0.5, out=None):
+    if width > 0.0:
+        with set_numpy_err(over="ignore"):
+            return np.divide(1.0, (1.0 + np.exp(-2.0 / width * np.asarray(x))), out=out)
+    else:
+        return np.heaviside(x, 0.5)
+
+
+class CarbonateProducer(Component):
+    """When I properly understand how to write the code, I'll write this intro ;-)"""
+
+    _name = "The Producer of Carbonate"

```

Comment:
To be consistent with other components, suggest using the same name as the class (CarbonateProducer)